### PR TITLE
Issue 112

### PR DESCRIPTION
**Pull request by jramadeu**
pull request related to #112

### CODE REVIEW

**Comment by albertohata** [comment]
The "Travis CI build could not complete", but not sure what is causing this problem. 
Do you know how to fix this problem @berezovskyi?

File: simulation-ros/src/turtlebot2i/turtlebot2i_moveit/launch/planning_context.launch [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<launch>
+  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
+  <arg name="load_robot_description" default="false"/>
+
+  <!-- The name of the parameter under which the URDF is loaded -->
+  <arg name="robot_description" default="robot_description"/>
+
+  <!-- Load universal robot description format (URDF) -->
+  <!-- TODO understand this file. For now it isn't launching -->
```
Comment (albertohata) [inline simulation-ros/src/turtlebot2i/turtlebot2i_moveit/launch/planning_context.launch:9]: Are these comments correct (line 8 and 9)?

Comment (jramadeu) [reply simulation-ros/src/turtlebot2i/turtlebot2i_moveit/launch/planning_context.launch:9]: Yes... As long as the arg `load_robot_description` is false robot's URDF is not loaded inside this file.

**Comment by albertohata** [comment]
**When launching [simulation-ros/src/turtlebot2i/turtlebot2i_bringup/launch/turtlebot2i_full.launch](https://github.com/EricssonResearch/scott-eu/pull/115/commits/f247c9bad2b4fa50646c33167ca65453631f6f40#diff-6340f3d50faab79700d0b54980ac583b), it is returning the following errors:** 

ERROR: cannot launch node of type [phantomx_pincher_arm_vrep_controller/controller_remoteApi]: can't locate node [controller_remoteApi] in package [phantomx_pincher_arm_vrep_controller]

RuntimeError: Unable to connect to move_group action server 'move_group' within allotted time (5s)        
RuntimeError: Unable to connect to move_group action server 'move_group' within allotted time (5s)        
[ERROR] [1538769941.366999145]: Action client not connected: phantom_controller/follow_joint_trajectory   
[ WARN] [1538769946.473850370]: Waiting for gripper_controller/gripper_cmd to come up                     
[rviz-12] process has died [pid 8765, exit code -11, cmd /opt/ros/kinetic/lib/rviz/rviz -d /home/alberto/scott-eu/simulation-ros/src/turtlebot2i/turtlebot2i_bringup/config/rviz_nav_arm.rviz __name:=rviz __log:=/home/alberto/.ros/log/e6db9266-c8d7-11e8-89e3-d46a6afe4763/rviz-12.log].                                   
log file: /home/alberto/.ros/log/e6db9266-c8d7-11e8-89e3-d46a6afe4763/rviz-12*.log                        
[ WARN] [1538769952.474014896]: Waiting for gripper_controller/gripper_cmd to come up                     
[WARN] [1538769954.305630]: Controller Spawner couldn't find the expected controller_manager ROS interface.                                                                                                         
[controller_manager_spawner-8] process has finished cleanly                                               
log file: /home/alberto/.ros/log/e6db9266-c8d7-11e8-89e3-d46a6afe4763/controller_manager_spawner-8*.log   
[ERROR] [1538769958.474141944]: Action client not connected: gripper_controller/gripper_cmd               
[ INFO] [1538769958.481821057]: Returned 0 controllers in list                                            
[ INFO] [1538769958.492394581]: Trajectory execution is managing controllers

@jramadeu, do you have an idea of can be causing this?

**Comment by albertohata** [comment]
**When launching [simulation-ros/src/turtlebot2i/turtlebot2i_moveit/launch/demo_full.launch](https://github.com/EricssonResearch/scott-eu/pull/115/commits/b7fd704ee5deb15e14503dfcb4cb104cfb517619#diff-1fe4ac79ec103e2b03ce7a45dabd25e4), it is returning the following errors:**

[ERROR] [1538769279.052522904]: Unable to identify any set of controllers that can actuate the specified joints: [ PhantomXPincher_joint1 PhantomXPincher_joint2 PhantomXPincher_joint3 PhantomXPincher_joint4 ]
[ERROR] [1538769279.052538188]: Known controllers and their joints:


ERROR: cannot launch node of type [phantomx_pincher_arm_vrep_controller/controller_remoteApi]: can't locate node [controller_remoteApi] in package [phantomx_pincher_arm_vrep_controller]

@jramadeu, do you have an idea of can be causing this?

**Comment by jramadeu** [comment]
Both messages are due to this:

> ERROR: cannot launch node of type [phantomx_pincher_arm_vrep_controller/controller_remoteApi]: can't locate node [controller_remoteApi] in package [phantomx_pincher_arm_vrep_controller]

After a proper source on the environment , you should be able to start controller_remoteApi from the package phantomx_pincher_arm_vrep_controller. If does not work, compile the package, source de env and try again.

**Comment by jramadeu** [comment]
@berezovskyi , can you help us with this CI build failure?

**Comment by albertohata** [comment]
> ERROR: cannot launch node of type [phantomx_pincher_arm_vrep_controller/controller_remoteApi]: can't locate node [controller_remoteApi] in package [phantomx_pincher_arm_vrep_controller]
> 
> RuntimeError: Unable to connect to move_group action server 'move_group' within allotted time (5s)
> RuntimeError: Unable to connect to move_group action server 'move_group' within allotted time (5s)
> [ERROR] [1538769941.366999145]: Action client not connected: phantom_controller/follow_joint_trajectory
> [ WARN] [1538769946.473850370]: Waiting for gripper_controller/gripper_cmd to come up
> [rviz-12] process has died [pid 8765, exit code -11, cmd /opt/ros/kinetic/lib/rviz/rviz -d /home/alberto/scott-eu/simulation-ros/src/turtlebot2i/turtlebot2i_bringup/config/rviz_nav_arm.rviz __name:=rviz __log:=/home/alberto/.ros/log/e6db9266-c8d7-11e8-89e3-d46a6afe4763/rviz-12.log].
> log file: /home/alberto/.ros/log/e6db9266-c8d7-11e8-89e3-d46a6afe4763/rviz-12*.log
> [ WARN] [1538769952.474014896]: Waiting for gripper_controller/gripper_cmd to come up
> [WARN] [1538769954.305630]: Controller Spawner couldn't find the expected controller_manager ROS interface.
> [controller_manager_spawner-8] process has finished cleanly
> log file: /home/alberto/.ros/log/e6db9266-c8d7-11e8-89e3-d46a6afe4763/controller_manager_spawner-8*.log
> [ERROR] [1538769958.474141944]: Action client not connected: gripper_controller/gripper_cmd
> [ INFO] [1538769958.481821057]: Returned 0 controllers in list
> [ INFO] [1538769958.492394581]: Trajectory execution is managing controllers

The timeout problem is caused by the slow frame rate of V-REP. Disabling the sensors may help to increase the frame rate and fix this problem, though it would be better in the future to detect automatically this problem and "respawn" the node when it happens.

**Review by albertohata — APPROVED** [review]

**Pull request merged**